src/macfloat.c

```diff
@@ -433,7 +433,7 @@ Obj FuncINTFLOOR_MACFLOAT( Obj self, Obj obj )
 
 Obj FuncSTRING_DIGITS_MACFLOAT( Obj self, Obj gapprec, Obj f)
 {
-  Char buf[50];
+  Char buf[1024];
   Obj str;
   int prec = INT_INTOBJ(gapprec);
   if (prec > 40) /* too much anyways, and would risk buffer overrun */
```

Comment:
Actually, while looking at this, I realized that this code could be called with a negative precision... Anyway, the problem here is that some GCC versions are not clever enough to detect that prec <= 40 holds after this, and hence issue that spurious warning, based on the maximal possible precision. This warning is not triggered for the first call to `PrintMacfloatToBuf`, because there the precision is specified as a hard coded constant (`16`).